### PR TITLE
[gatsby] Remove "gatsby" dependency, use typebox instead of ajv for validation

### PR DESCRIPTION
**Pull request by TooTallNate**
The "gatsby" dependency is large and relies on native packages ("sharp" specifically, which are error prone because it attempts to download a precompiled binary, of which the CDN has been down before causing us to not be able to merge anything).

The package is only used for TypeScript types, and we've decided that including the package only for those types is not worth the trouble. The existing JSON schemas already validate at runtime that the Gatsby redux store provides the values that we depend on.

`ajv` was ditched in favor of TypeBox types, which provide both JSON Schema and validation using a simpler syntax.

### CODE REVIEW

File: packages/gatsby-plugin-vercel-builder/package.json [hunk 1]
```diff
@@ -14,10 +14,10 @@
     "build:src": "tsc -p tsconfig.src.json"
   },
   "dependencies": {
+    "@sinclair/typebox": "0.25.24",
     "@vercel/build-utils": "6.3.2",
     "@vercel/node": "2.9.10",
     "@vercel/routing-utils": "2.1.10",
-    "ajv": "8.12.0",
```
Comment (styfle) [inline packages/gatsby-plugin-vercel-builder/package.json:20]: I still see ajv being imported in `packages/gatsby-plugin-vercel-builder/src/schemas.ts`

Did you intentionally to remove it from package.json?

Comment (TooTallNate) [reply packages/gatsby-plugin-vercel-builder/package.json:20]: Removed now: 3608e8a

**Review by cb1kenobi — DISMISSED** [review]
LGTM

File: packages/gatsby-plugin-vercel-builder/src/schemas.ts [hunk 2]
```diff
@@ -1,98 +1,52 @@
...
+  trailingSlash: Type.Optional(
+    StringEnum(['always', 'never', 'ignore', 'legacy'])
+  ),
+});
+export type GatsbyConfig = Static<typeof GatsbyConfigSchema>;
+
+const GatsbyStateSchema = Type.Object({
+  pages: Type.Array(Type.Tuple([Type.String(), GatsbyPageSchema])),
+  redirects: Type.Array(GatsbyRedirectSchema),
+  functions: Type.Array(GatsbyFunctionSchema),
```
Comment (styfle) [inline packages/gatsby-plugin-vercel-builder/src/schemas.ts:47]: Do we have tests covering all of these config inputs?

**Review by styfle — APPROVED** [review]

File: packages/gatsby-plugin-vercel-builder/src/schemas.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/gatsby-plugin-vercel-builder/src/schemas.ts:47]: Just the e2e tests at the moment.

**Pull request merged**